src/mca/common/dstore/dstore_base.c

```diff
@@ -2695,10 +2761,12 @@ PMIX_EXPORT pmix_status_t pmix_common_dstor_store_modex(pmix_common_dstore_ctx_t
             PMIX_DESTRUCT(&pbkt);
             return rc;
         }
-        if (PMIX_SUCCESS != (rc = pmix_common_dstor_store(ds_ctx, &proc, PMIX_REMOTE, kv))) {
+
+        if (PMIX_SUCCESS != (rc = pmix_common_dstor_store(ds_ctx, &proc, PMIX_REMOTE, kv, true))) {
```

Comment:
Since the `_mdxcbfunc` always sets `holding_ns_locks = true`: why we cannot introduce `pmix_common_dstor_store_nolock` here,  it allows to avoid add the new argument for `pmix_common_dstor_store` and `pmix_common_dstor_store_modex`? Also in this way, `pmix_common_dstor_store_modex` may not use locks.
I think it reduces the commit changes.